### PR TITLE
Search for submittable elements recursively

### PR DESCRIPTION
**Pull request by TomK**
This PR improves the submittable element detection to include deeply nested and wrapped elements.

### CODE REVIEW

File: iron-form.html [hunk 1]
```diff
@@ -361,23 +361,30 @@
       },
 
       _getSubmittableElements: function() {
-        var nodes = Polymer.dom(this._form).children;
+        return this._findElements(this._form);
```
Comment (notwaldorf) [inline iron-form.html]: Isn't this also needed for `getValidatableElements` ?

Comment (notwaldorf) [inline iron-form.html]: Also, the `Polymer.dom(this._form).children` line has changed :(

**Comment by TomK** [comment]
Thanks for the feedback @notwaldorf.  I have made the reqested changes now.

Could you confirm please, `_getValidatableElements` does not require an element to have the `name` property, but `_getSubmittableElements` does.
I think this means we will validate every element, input or otherwise. Is that by design?

**Comment by notwaldorf** [comment]
Yup. If you look at a native form example (http://jsbin.com/rulenig/edit?html,output), forms validate elements without a `name` attribute, but will only submit the ones with a `name` attribute

**Comment by TomK** [comment]
Ah great. Thanks for explaining.  This PR should match that then :)

File: iron-form.html [hunk 2]
```diff
@@ -355,33 +355,31 @@
...
+          var node = nodes[i];
           // An element is submittable if it is not disabled, and if it has a
           // 'name' attribute.
-          if (!nodes[i].disabled && nodes[i].name) {
-            submittable.push(nodes[i]);
+          if(!node.disabled && (ignoreName || node.name)) {
+            submittable.push(node);
+          }
+          else {
+            // This element might contain other potentially submittable elements
```
Comment (notwaldorf) [inline iron-form.html]: is this comment correct?

Comment (TomK) [inline iron-form.html]: should be now 👍

**Comment by notwaldorf** [comment]
This is awesome! Thanks so much for putting up with the review!

**Pull request merged**